### PR TITLE
Dp 690 create stub api

### PR DESCRIPTION
**Pull request by drEfrati**
#### Main purpose (bug/feature/enhancement + description)
 To add the stub api

### CODE REVIEW

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/KinAccount.java [hunk 1]
```diff
@@ -100,7 +99,7 @@
      * @throws OperationFailedException other error occurred.
      */
     @NonNull
-    TransactionId sendTransactionSync(Transaction transaction) throws OperationFailedException;
+    TransactionId sendTransactionSync(PaymentTransaction transaction) throws OperationFailedException;
 
     /**
      * send a whitelist transaction.
```
[the remark this answers is not in the document]
Comment (drEfrati) [reply kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/KinAccount.java]: Will add them in next task

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/exception/DecodeTransactionException.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package kin.sdk.exception;
+
+public class DecodeTransactionException extends Exception {
```
[the remark this answers is not in the document]
Comment (drEfrati) [reply kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/exception/DecodeTransactionException.java]: Ok, and I will change the docs regarding this change

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/queue/PaymentQueue.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     * @param fee the amount of fee(in quarks) for each payment
+     */
+    void setFee(int fee);
+
+    /**
+     * provide information on this queue
+     *
+     * @return the status pf this queue.
+     */
+    Status status(); // TODO: 2019-08-04 this is a bit weird to return an object which has 2 methods to query some
```
[the remark this answers is not in the document]
Comment (drEfrati) [reply kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/queue/PaymentQueue.java]: never mind I will leave it as is, maybe we will add more statuses

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/transaction_data/Transaction.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    public static RawTransaction decodeRawTransaction(String transactionEnvelope) throws DecodeTransactionException {
+        try {
+            kin.base.Transaction transaction = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+            return new RawTransaction(transaction);
+        } catch (IOException e) {
+            throw new DecodeTransactionException(e.getMessage(), e.getCause());
+        }
+    }
+
+    public kin.base.Transaction baseTransaction() {
```
[the remark this answers is not in the document]
Comment (drEfrati) [reply kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/transaction_data/Transaction.java]: Will keep it public for now until a further investigation to what will be the best approach in this case

**Comment by drEfrati** [comment]
@yosriz handled all the comments so please check again

File: kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/transactiondata/Transaction.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    public static RawTransaction decodeRawTransaction(String transactionEnvelope) throws DecodeTransactionException {
+        try {
+            kin.base.Transaction transaction = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+            return new RawTransaction(transaction);
+        } catch (IOException e) {
+            throw new DecodeTransactionException(e.getMessage(), e.getCause());
+        }
+    }
+
+    public kin.base.Transaction baseTransaction() {
```
Comment (yosriz) [inline kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/transactiondata/Transaction.java]: hide by introducing a middle level class that will be used only internally (TransactionBase)

**Pull request merged**